### PR TITLE
[ci] Add upload of test results to JFrog (#7378)

### PR DESCRIPTION
**Pull request by aHenryJard**
### Proposed changes

* Add a drone step to upload frontend test artefact to JFrog for analyses on failures that run on both success and failure of e2e step.

I did not use the drone plugin for JFrog because it's not customizable enough, I used jfrog cli instead.

Build artefact are visible on https://filigran.jfrog.io/ui/builds/

Example with build 22529 (with one e2e fail in purpose)
image

Jfrog URL can be found on upload artifact step:
image

And in artifact there is screenshot and video from failure
image

### Related issues
* Relates to #7378 work, does not close it.

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case (coverage and e2e)
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -102,6 +102,35 @@ steps:
...
+      # Archive and upload each artefact that we need
+      #- tar -czvf frontend-test-results.tar.gz opencti-platform/opencti-front/test-results
+      - tar -czvf frontend-test-results-$DRONE_BUILD_NUMBER.tar.gz opencti-platform/opencti-front/tests_e2e
+      - jf rt u frontend-test-results-$DRONE_BUILD_NUMBER.tar.gz $JFROG_REPOSITORY --build-name=$JFROG_BUILD_NAME --build-number=$DRONE_BUILD_NUMBER --url=$JFROG_URL --access-token=$JFROG_TOKEN
+      # Next line should be done only once at the end: it's recording and gathering build info
+      - jf rt bp $JFROG_BUILD_NAME $DRONE_BUILD_NUMBER --url=$JFROG_URL --access-token=$JFROG_TOKEN --build-url=$DRONE_BUILD_LINK
+      # Cleaning up old build in JFrog
+      - jf rt bdi $JFROG_BUILD_NAME --max-days=30 --url=$JFROG_URL --access-token=$JFROG_TOKEN
+    #depends_on:
+    #  - frontend-e2e-tests
```
Comment (aHenryJard) [inline .drone.yml]:
```suggestion
    depends_on:
      - frontend-e2e-tests
```

File: .drone.yml [hunk 2]
```diff
@@ -102,6 +102,35 @@ steps:
...
+      - echo "CI = $CI"
+      - apt-get update
+      # see https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory
+      - npm install -g jfrog-cli-v2-jf
+      - set -x
+      # Collect git info
+      - jf rt bag $JFROG_BUILD_NAME $DRONE_BUILD_NUMBER
+      # Archive and upload each artefact that we need
+      #- tar -czvf frontend-test-results.tar.gz opencti-platform/opencti-front/test-results
+      - tar -czvf frontend-test-results-$DRONE_BUILD_NUMBER.tar.gz opencti-platform/opencti-front/tests_e2e
```
Comment (aHenryJard) [inline .drone.yml]:
```suggestion
      - tar -czvf frontend-test-results.tar.gz opencti-platform/opencti-front/test-results
```

File: opencti-platform/opencti-front/tests_e2e/navigation/navigation.spec.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ const navigateReports = async (page: Page) => {
 
   // -- Knowledge
   await reportDetailsPage.goToKnowledgeTab();
-  await expect(page.getByTestId('report-knowledge')).toBeVisible();
+  await expect(page.getByTestId('report-knowledge-fail')).toBeVisible();
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/tests_e2e/navigation/navigation.spec.ts]:
```suggestion
  await expect(page.getByTestId('report-knowledge')).toBeVisible();
```

File: .drone.yml [hunk 4]
```diff
@@ -102,6 +102,38 @@ steps:
...
+        - success
+    environment:
+      JFROG_TOKEN:
+          from_secret: jfrog_token
+      JFROG_BUILD_NAME: opencti-build
+      JFROG_REPOSITORY: opencti-drone
+      JFROG_URL: https://filigran.jfrog.io/artifactory
+    commands:
+      - echo "DRONE_BUILD_NUMBER = $DRONE_BUILD_NUMBER"
+      - echo "CI = $CI"
```
Comment (aHenryJard) [inline .drone.yml]:
```suggestion
```

File: .drone.yml [hunk 5]
```diff
@@ -102,6 +102,38 @@ steps:
...
+      - apt-get update
+      # see https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory
+      - npm install -g jfrog-cli-v2-jf
+      - set -x
```
Comment (aHenryJard) [inline .drone.yml]:
```suggestion
```

File: opencti-platform/opencti-front/playwright.config.ts [hunk 6]
```diff
@@ -31,10 +31,11 @@ export default defineConfig({
         entryFilter: (entry) => true,
         sourceFilter: (sourcePath) => sourcePath.startsWith('src'),
       },
+      /*
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/playwright.config.ts:34]: Are we keeping it for reference or should it be removed ?

**Review by lndrtrbn — APPROVED** [review]

File: opencti-platform/opencti-front/playwright.config.ts (hunk 6 above)
Comment (aHenryJard) [reply opencti-platform/opencti-front/playwright.config.ts:34]: yes I'm keeping it in case we want to enable it again later, for now it's not used

**Pull request merged**